packages/editor/src/lib/editor/managers/LicenseManager.ts

```diff
@@ -68,6 +93,33 @@ export class LicenseManager {
 		}
 	}
 
+	private getExpirationDateWithGracePeriod(expiryDate: Date) {
+		return new Date(
+			expiryDate.getFullYear(),
+			expiryDate.getMonth(),
+			expiryDate.getDate() + GRACE_PERIOD_DAYS + 1 // Add 1 day to include the expiration day
+		)
+	}
+
+	private isAnnualLicenseExpired(expiryDate: Date) {
+		const expiration = this.getExpirationDateWithGracePeriod(expiryDate)
+		return new Date() >= expiration
+	}
+
+	private isPerpetualLicenseExpired(expiryDate: Date) {
+		const expiration = this.getExpirationDateWithGracePeriod(expiryDate)
```

Comment:
Hm, not sure if we want to use the grace period for perpetual licenses. Feels strange since it's comparing to release dates. I'd probably remove it?